Install firewalld and libvirt-dbus on Debian/Ubuntu images

martinpitt opened the pull request:
See individual commits for details.

 * [x] image-refresh debian-stable
 * [x] image-refresh debian-testing
 * [x] image-refresh ubuntu-1804
 * [x] image-refresh ubuntu-stable
 * [x] [TestMachinesDBus.testLibvirt](https://fedorapeople.org/groups/cockpit/logs/pull-10512-20181107-044848-406009d3-verify-ubuntu-stable/log.html#241) on ubuntu-stable
 * [x] `TestMachines{Virsh,DBus}.testBasic` failure on debian-testing and ubuntu-stable
 * [x] [TestDocker.testExpose](https://fedorapeople.org/groups/cockpit/logs/pull-10512-20181107-035858-406009d3-verify-debian-stable/log.html#29) on debian-stable
 * [x] [testNetworkSettings](https://fedorapeople.org/groups/cockpit/logs/pull-10512-20181107-210303-bc54fcd8-verify-debian-testing/log.html#219)
 * [x] [testNetworkingPage](https://fedorapeople.org/groups/cockpit/logs/pull-10512-20181108-034545-51ace850-verify-debian-testing/log.html#249) on debian-testing
 * [x] [check-machines-dbus](https://fedorapeople.org/groups/cockpit/logs/pull-10512-20181108-033030-51ace850-verify-ubuntu-stable/log.html#228) on ubuntu-stable

cockpituous commented:
image-refresh in progress on bos-tasks-50pch.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-10512-20181106-110101/

cockpituous commented:
image-refresh debian-testing done: https://github.com/cockpituous/cockpit/commits/image-refresh-debian-testing-20181106-121313

cockpituous commented:
image-refresh in progress on bos-tasks-80p4z.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-10512-20181106-121414/

cockpituous commented:
image-refresh ubuntu-1804 done: https://github.com/cockpituous/cockpit/commits/image-refresh-ubuntu-1804-20181106-123838

cockpituous commented:
image-refresh in progress on bos-tasks-j7k1l.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-10512-20181106-125050/

cockpituous commented:
image-refresh ubuntu-stable done: https://github.com/cockpituous/cockpit/commits/image-refresh-ubuntu-stable-20181106-132222

martinpitt commented:
It clearly [doesn't like](https://fedorapeople.org/groups/cockpit/logs/pull-10512-20181106-143535-c6dd7645-verify-ubuntu-stable/log.html#238) libvirt-dbus on ubuntu-stable.

martinpitt commented:
Huh, what?

```
root@ibm-p8-kvm-03-guest-02:~# busctl call org.libvirt /org/libvirt/QEMU org.libvirt.Connect ListDomains u 0
Failed to connect socket to '/var/run/libvirt/libvirt-sock': Permission denied

# ls -l /var/run/libvirt/libvirt-sock
srwxrwx--- 1 root libvirt 0 Nov  6 15:15 /var/run/libvirt/libvirt-sock

# ps aux|grep libvirtd
root      1434  0.0  3.4 1038028 35212 ?       Ssl  15:15   0:00 /usr/sbin/libvirtd
```
@KKoukiou , any idea here?

martinpitt commented:
@KKoukio: /var/run/libvirt/libvirt-sock has permissions 777 on Fedora; when I chmod it, it works.

martinpitt commented:
@KKoukiou : On debian-testing, the socket is also 777. It's the exact same libvirt-dbus version, hmm.

I filed https://bugs.launchpad.net/bugs/1802005 about it.

martinpitt commented:
The [testExpose](https://fedorapeople.org/groups/cockpit/logs/pull-10512-20181107-035858-406009d3-verify-debian-stable/log.html#29) failure on debian-stable seems iptables related:

```
docker[723]: [fc61a65e.allocate_port(6a78d63c467f2813bbcfd52983db04d954fb64eab89067dbad34529ed1b5cd22)] Failed to allocate and map port 3380: iptables failed: iptables --wait -t nat -A DOCKER -p tcp -d 0/0 --dport 3380 ! -i docker0 -j DNAT --to-destination 172.17.0.8:3380: iptables: No chain/target/match by that name.
Nov 07 02:16:38 unassigned-hostname docker[723]:  (exit status 1)
Nov 07 02:16:38 unassigned-hostname docker[723]: iptables failed: iptables --wait -t nat -A DOCKER -p tcp -d 0/0 --dport 3380 ! -i docker0 -j DNAT --to-destination 172.17.0.8:3380: iptables: No chain/target/match by that name.
Nov 07 02:16:38 unassigned-hostname docker[723]:  (exit status 1)
```
There is indeed no "docker" chain.

This is reproducible by a trivial `docker run -it --rm -p 3380:3380 busybox /bin/sh`.

However, there is no bug to report downstream, as we are using a totally unsupported package: There is no official docker Debian package for Debian 9, we use the backport for Debian 8. We just have to accept that this does not work with firewalld. So I'll add a test hack to stop firewalld for that test.

martinpitt commented:
And finally the [TestMachinesDBus.testBasic](https://fedorapeople.org/groups/cockpit/logs/pull-10512-20181107-043535-406009d3-verify-debian-testing/log.html#213) failure: This is apparently independent from libvirt-dbus, as it also happens with virsh.

When running with `-v`, the booted inner VM indeed doesn't arrive at the login prompt, but fails with a kernel panic:
```
[    0.176000] Initializing cgroup subsys pids
[    0.184000] mce: CPU supports 10 MCE banks
[    0.188000] Last level iTLB entries: 4KB 0, 2MB 0, 4MB 0
[    0.192000] Last level dTLB entries: 4KB 0, 2MB 0, 4MB 0, 1GB 0
[    0.468000] Freeing SMP alternatives memory: 28K (c1c69000 - c1c70000)
[    0.484000] ftrace: allocating 31209 entries in 61 pages
[    0.580000] smpboot: Max logical packages: 1
[    0.584000] smpboot: APIC(0) Converting physical 0 to logical package 0
[    0.588000] Enabling APIC mode:  Flat.  Using 1 I/O APICs
[    0.600000] ..TIMER: vector=0x30 apic1=0 pin1=2 apic2=-1 pin2=-1
[    0.600000] ..MP-BIOS bug: 8254 timer not connected to IO-APIC
[    0.600000] ...trying to set up timer (IRQ0) through the 8259A ...
[    0.600000] ..... (found apic 0 pin 2) ...
[    0.600000] ....... failed.
[    0.600000] ...trying to set up timer as Virtual Wire IRQ...
[    0.616000] ..... failed.
[    0.616000] ...trying to set up timer as ExtINT IRQ...
[    0.620000] ..... failed :(.
[    0.620000] Kernel panic - not syncing: IO-APIC + timer doesn't work!  Boot with apic=debug and send a report.  Then try booting with the 'noapic' option.
[    0.620000] 
[    0.620000] CPU: 0 PID: 1 Comm: swapper/0 Not tainted 4.4.0-28-generic #47-Ubuntu
[    0.620000] Hardware name: QEMU Standard PC (i440FX + PIIX, 1996), BIOS 1.11.1-1ubuntu1 04/01/2014
[    0.620000]  c1ac3967 f428ecbb 00000086 cf0a7ec8 c13a05df 00000000 00000008 cf0a7ee0
[    0.620000]  c116fb30 00000008 00000000 00000008 00000000 cf0a7f3c c1b930b0 c19ae6a0
[    0.620000]  00000246 0000000f 00000001 fffffffc 00010000 00000000 00000026 00000002
[    0.620000] Call Trace:
[    0.620000]  [<c13a05df>] dump_stack+0x58/0x79
[    0.620000]  [<c116fb30>] panic+0x81/0x1af
[    0.620000]  [<c1b930b0>] check_timer+0x56c/0x5a0
[    0.620000]  [<c1b93a47>] setup_IO_APIC+0x178/0x18a
[    0.620000]  [<c1b920a1>] apic_bsp_setup+0x98/0xaa
[    0.620000]  [<c1b9004f>] native_smp_prepare_cpus+0x2a4/0x316
[    0.620000]  [<c1b7cc70>] kernel_init_freeable+0x97/0x1d4
[    0.620000]  [<c17a1b50>] kernel_init+0x10/0xe0
[    0.620000]  [<c1099571>] ? schedule_tail+0x11/0x50
[    0.620000]  [<c17ab7c9>] ret_from_kernel_thread+0x21/0x38
[    0.620000]  [<c17a1b40>] ? rest_init+0x70/0x70
[    0.620000] ---[ end Kernel panic - not syncing: IO-APIC + timer doesn't work!  Boot with apic=debug and send a report.  Then try booting with the 'noapic' option.
[    0.620000] 
```
However, when I run `virsh start subVmTest1`, I see in /var/log/libvirt/console-subVmTest1.log not a kernel panic, but that it hangs on networking:
```Initializing random number generator... done.
Starting acpid: OK
Starting network...
udhcpc (v1.23.2) started
Sending discover...
Sending discover...
```
After a long timeout, it finally arrives at the cirros login prompt.

Booting with the default network works fine:

    qemu-system-x86_64 -enable-kvm -nographic -hda /dev/sda -snapshot

Booting without network *also* boots fast:

    qemu-system-x86_64 -enable-kvm -nographic -hda /dev/sda -snapshot  -net none

Then only lo is configured, and eth are skipped. So it seems the integration test starts the VM with existing network cards but no DHCP server. The test already allows a 180s timeout, but the "discover..." timeout takes longer than that, about 3:30 minutes.

So at this point I'm not sure if on production CI we actually see the kernel panic or the DHCP timeout. I ran the test again, and that time it indeed *was* the network timeout.

On e. g. debian-stable, the cirros VMs start really fast, and they *do* get a DHCP response:
```
=== network info ===
if-info: lo,up,127.0.0.1,8,,
if-info: eth0,up,192.168.122.35,24,fe80::5054:ff:fed4:e366/64,
ip-route:default via 192.168.122.1 dev eth0 
ip-route:192.168.122.0/24 dev eth0  src 192.168.122.35 
```
So it seems the host-side DHCP server from libvirt is broken there.

martinpitt commented:
> So it seems the host-side DHCP server from libvirt is broken there.

It's just because firewalld is now running, and we don't configure it to allow traffic on virbr0. This isn't an issue on e. g. the Fedora or RHEL images as they don't enable firewalld by default.

martinpitt commented:
[This failure](https://fedorapeople.org/groups/cockpit/logs/pull-10512-20181107-210303-bc54fcd8-verify-debian-testing/log.html#219) also looks real.

martinpitt commented:
<details><summary>Pull request has been rewritten.</summary>

[Old commits](https://github.com/cockpit-project/cockpit/commits/bc54fcd8de30703cfd646b7c662bd50bec2b7e52) | [Diff](https://github.com/cockpit-project/cockpit/compare/bc54fcd8de30703cfd646b7c662bd50bec2b7e52..51ace85067939c72bd89d1512e2d6b991f215f99)
```diff
diff --git a/bots/images/candlepin b/bots/images/candlepin
index 875b76ea1..58ceac1dd 120000
--- a/bots/images/candlepin
+++ b/bots/images/candlepin
@@ -1 +1 @@
-candlepin-38bb32716cb44da5202e2bff2921203416eedb6c24d09b9b25fec237273af61a.qcow2
\ No newline at end of file
+candlepin-44efad7b05562a81cfdc417dfd2b5d8b80d7beda84c9c9ecb1b7a46d6463ce7d.qcow2
\ No newline at end of file
diff --git a/bots/images/files/openshift.kubeconfig b/bots/images/files/openshift.kubeconfig
index 163a98149..82a333741 100644
--- a/bots/images/files/openshift.kubeconfig
+++ b/bots/images/files/openshift.kubeconfig
@@ -1,7 +1,7 @@
 apiVersion: v1
 clusters:
 - cluster:
-    certificate-authority-data: LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUM2akNDQWRLZ0F3SUJBZ0lCQVRBTkJna3Foa2lHOXcwQkFRc0ZBREFtTVNRd0lnWURWUVFEREJ0dmNHVnUKYzJocFpuUXRjMmxuYm1WeVFERTFNemczTVRFME16TXdIaGNOTVRneE1EQTFNRE0xTURNeVdoY05Nak14TURBMApNRE0xTURNeldqQW1NU1F3SWdZRFZRUUREQnR2Y0dWdWMyaHBablF0YzJsbmJtVnlRREUxTXpnM01URTBNek13CmdnRWlNQTBHQ1NxR1NJYjNEUUVCQVFVQUE0SUJEd0F3Z2dFS0FvSUJBUUNzRndRaGxOMW5ycDRVQ0tNaEtZY3YKUjVBMW5LdnNyMFdIMFlpaXdEVHNNZ3E2QmgyREU0VjBRTS94SWlHUi9MNGR6RzhmNVZXUytDTVRFVVhhTjlheQpLSjRTR2h5ekQ0K1llUjVsakM4c0VYWEx1RmYybnlmZnRIdll4SG9wcjU4WHJkNndQTVMrbVJESmc1bHg4UXVHCjhaNWZwYlgyZnd6MjY0Vnp0SFdwOUZsYUp2VzFaVGg0ZXhsN1RWSnErVGtjUG5lYUE0RkRjMzV1TEkrdEY0cHEKTVcwTmZzRytTMDB2UXVUMnJHci9KaEs3TjVsOUs0azdPbUs1eEhaWlExWlNSNFEzRGN2RFFXS2kwUHgxeUpONwpLMHpJaWVoeXdaeVZBWEYrQ3FJbEo4cmR4Sit2dk9zRXUvU05YSkErc2h4Vy8xYWV5U1c3emZwSUxSaUJ3dEZKCkFnTUJBQUdqSXpBaE1BNEdBMVVkRHdFQi93UUVBd0lDcERBUEJnTlZIUk1CQWY4RUJUQURBUUgvTUEwR0NTcUcKU0liM0RRRUJDd1VBQTRJQkFRQ0FMZGg5akE1bmN1aUFIYkZQcVBXM1VNT0VEMUVmZDZRZFZLdjc5VWFYNEhDOApra2VpQWtYSWg5KzgzaTBRdFRZL1Q5SGRKdDlQekFKM0NIbElrNFRLb1NkMHVYQitvblo3eW81RDFwVTArS3VWCjh5N2RnaXlmUnA3RWJsM1MwQnlQVHYrSnlYSHE2QTdDRXptYlBuNUJ2N3MzNkxrOGJ2Y1E5NlVYN0crSmRUaWsKd0wrYjIyQlFVeExWWFI4Zk1GOUxDSzBZNzJnTkdBZkJVTzVvVXY4YmZWVHZ0R2tGRmF1bkwxdmc3aWVORDU3bgpaOEx2SE5GMWtscnNuNkhpNUY4NkZDcWl0bFhBa0ZrY1c3NEpIb1JqYUs1WGY0cnN3b3VEQTAwQldoM0RtUUtDCjdvWWRTczF1bzRaMHFNdW9PSnF5OWE3MVJibHlxM0JhUUpUa1BrejQKLS0tLS1FTkQgQ0VSVElGSUNBVEUtLS0tLQo=
+    certificate-authority-data: LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUM2akNDQWRLZ0F3SUJBZ0lCQVRBTkJna3Foa2lHOXcwQkFRc0ZBREFtTVNRd0lnWURWUVFEREJ0dmNHVnUKYzJocFpuUXRjMmxuYm1WeVFERTFOREUxTlRBM05EWXdIaGNOTVRneE1UQTNNREF6TWpJMVdoY05Nak14TVRBMgpNREF6TWpJMldqQW1NU1F3SWdZRFZRUUREQnR2Y0dWdWMyaHBablF0YzJsbmJtVnlRREUxTkRFMU5UQTNORFl3CmdnRWlNQTBHQ1NxR1NJYjNEUUVCQVFVQUE0SUJEd0F3Z2dFS0FvSUJBUUM3d1JMWFRTZFBvWldaQzJuSGRNNkEKODYwRzZGZG95TWl0TnNDdXNZVXB2VitPWldpM2NON2dodXR2K3o4aWpkZUFWVm8xYkFrMUMyV3g3Nm1PRGJiSgpQY0JuVm9IZnpFVVhMMjZPclA3Q25KaTRrNitJSzBXZEVLOHhQdFprZDcyWFMvclpKd0NKT1hDcmg5TUorKzV1CjluaHJ1elNQNFhqaCsvcXJNUlNkNGx4RWM4MDRuMVUzYzFBUUNwc2V3UjhmbDRUQlMrTXNrejhvOFpsL1R6WksKcGNZYWQyZzZROFBnbTNYazRSaTF2Q0ZtQmpNVTRVQmRjU0g4VkR6YThxRlNzYUxrUzZvVjZlRmVodUpoTVJxSwpZOVVhdWxrd0ZERHdVQitYUWI4VzFmeU5ORlhySGxoczQrNENXR0hHSXU0c0h6RU1uUlBEcStsMWxnZ3R6enRaCkFnTUJBQUdqSXpBaE1BNEdBMVVkRHdFQi93UUVBd0lDcERBUEJnTlZIUk1CQWY4RUJUQURBUUgvTUEwR0NTcUcKU0liM0RRRUJDd1VBQTRJQkFRQWFKU05KVjFwajhRKzg4Vmp0UUV6a0FjWmpickxGN0VXNURhd2FyOS9Wc0RFbAp3OVoxSzlmeTdHeGRFdldudTRhOEtVZHpSS1pwUUU5TGtuOUlNVFI3ejBpTjdpMzlYUmdudmVWNi9PQyttUmE1CmRaek5VaTcyYXRVS1ZPa0x3RG9ZcXoxdG5ocEl0OU5SRDJmNWhRdWI0QUFZSHU2MWliTWFkMXJiNEtoMDkzZEoKUEZLMkdqYmZCK1FPbXRvN1JOQy9heVVwaGFaNWc0SnpTbUpHeFFVaG1BaWhoRUdVUEZMNWZFZVNUMnRTbnE0NQpjZkJqSUh1b2NhNlJjNG5idE1ZeGJjOFpsS2ZvRkdqeXhJc2dzd1B2M01icTl3c1hlcERmL3l1L1RhNnpLYjRxCnlzbFFnam4zS0FKL0M0VjZML1FOOTZBVVFKb3k2WCtKd1JGMnVYMkcKLS0tLS1FTkQgQ0VSVElGSUNBVEUtLS0tLQo=
     server: https://10.111.112.101:8443
   name: 10-111-112-101:8443
 contexts:
@@ -30,8 +30,8 @@ preferences: {}
 users:
 - name: scruffy/10-111-112-101:8443
   user:
-    token: owH5jKG4OglLgkCn7zL8SANt0_pbjvXdwaUgl97jufo
+    token: Bdccla5uVJXGaccJI_u44cl1Ck1Hnp0ijfVC11VKmqU
 - name: system:admin/10-111-112-101:8443
   user:
-    client-certificate-data: LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUM4akNDQWRvQ0NRQ3BueVVlR0YycUdEQU5CZ2txaGtpRzl3MEJBUXNGQURBbU1TUXdJZ1lEVlFRRERCdHYKY0dWdWMyaHBablF0YzJsbmJtVnlRREUxTXpnM01URTBNek13SGhjTk1UZ3hNREExTURNMU1UQTRXaGNOTWpBeApNREEwTURNMU1UQTRXakJRTVJjd0ZRWURWUVFLREE1emVYTjBaVzA2YldGemRHVnljekVlTUJ3R0ExVUVDZ3dWCmMzbHpkR1Z0T21Oc2RYTjBaWEl0WVdSdGFXNXpNUlV3RXdZRFZRUUREQXh6ZVhOMFpXMDZZV1J0YVc0d2dnRWkKTUEwR0NTcUdTSWIzRFFFQkFRVUFBNElCRHdBd2dnRUtBb0lCQVFDN29uMlhGVHdmUXg0TGRRWFBQOC9ZWUpNbwozUWI2MDQvRzZsd0sydTZYOStLVEd3QnJLSDQ1TWRpRFRuN2IxTEJyUW1peDlpdDBmS3lGbEh4cEwyb21NWTJGCm5Ycks4aVd5K1RSaTFhZzR1dXNZR3JNZTlkQlNxdGZ0endTdXh0Z28vVzU4TGVDVVdsNkdyUWxZSzc0SDdTdHQKUjhEOFRDb1lDY1ROZm1hNy9xbWtlRm5GUDRYVDZvK0lzMFU1TThwRURLTy9tL09VYkhKZ29USEJadFZNZTR5SgpaTFMyRGNKb01NYWVvT0pLN2JFSWpMWWRmY01NOWFvbUJqSUgzVnZFUVljU1dxUHFOQ0Q0cXJUWGJxWUJQSVdOCjgxYU4zTTlaa2Q5QTJhbG8zZlNIUDFrL212UmdDdjZFNUhhMnBRRWRLZTA5M2NjZEQ0dllYL01PSGREMUFnTUIKQUFFd0RRWUpLb1pJaHZjTkFRRUxCUUFEZ2dFQkFITCtHd1pIZEVRRjZ3MGd6bVppV0NmL1ZkRmZ2MzBXNUZ6agpraTBZb0ZLd00zNEVXeXJNS1JHZ1I4L3NzYmRGMC9ERGNPS2ZDVUtvOGx0TVFrUDIzazhQQlVydy9Pb1FJWHk5CkJKZnBFTDRjenpBTkpxMnIxSVVVMXVHUUN0WS9mU1lvdFFFa1pabUJCeVA0bFhLQkxtTU5WaU00aEoxRWJzTlcKTDNZMWNZRDlISW5HYURPSGpndmhmbkhXV0czc0J4V1p2VE14NXE0M3JGbC9jTjBHeE1veEpFS0Z5cEJzcXBQTQpGdDdiOXdQQ0NDU0w0ZjVyYnF2LytBTUF1cVp3amZRb3ZKSEdtY1VrOEV2elhOSjVjdUx6Y0dLdGlzeUc0WXVlCjhoQk1WdnVKVEVJVFpEc211dzVKSHNCbzlqSXp3bEZwUnNxUmVJM2ZzY0k2dms3YktIND0KLS0tLS1FTkQgQ0VSVElGSUNBVEUtLS0tLQo=
-    client-key-data: LS0tLS1CRUdJTiBSU0EgUFJJVkFURSBLRVktLS0tLQpNSUlFcFFJQkFBS0NBUUVBdTZKOWx4VThIME1lQzNVRnp6L1AyR0NUS04wRyt0T1B4dXBjQ3RydWwvZmlreHNBCmF5aCtPVEhZZzA1KzI5U3dhMEpvc2ZZcmRIeXNoWlI4YVM5cUpqR05oWjE2eXZJbHN2azBZdFdvT0xyckdCcXoKSHZYUVVxclg3YzhFcnNiWUtQMXVmQzNnbEZwZWhxMEpXQ3UrQiswcmJVZkEvRXdxR0FuRXpYNW11LzZwcEhoWgp4VCtGMCtxUGlMTkZPVFBLUkF5anY1dnpsR3h5WUtFeHdXYlZUSHVNaVdTMHRnM0NhRERHbnFEaVN1MnhDSXkyCkhYM0REUFdxSmdZeUI5MWJ4RUdIRWxxajZqUWcrS3EwMTI2bUFUeUZqZk5XamR6UFdaSGZRTm1wYU4zMGh6OVoKUDVyMFlBcitoT1IydHFVQkhTbnRQZDNISFErTDJGL3pEaDNROVFJREFRQUJBb0lCQVFDMitJbEgzeWsxd3hsMQpTclJUbVlKeEdodm02VUo1TGYweW1DOFJWdFQ3MUhwWndTYnpXQlZlYU1FZHFsb0N2TEVjcHpHNCtMai9hdVJCCmpyNlRRKzFiT1U4c1VKa0Jkb2hVc3JibStqcUV0Nkd4YTdmb0pRVDAyQ3lnZGhPMjY2aTJzQXdoaEF0UUg1KzEKN044RDRBUDdic2UrVlE1WCtWM0MreU95eitRV3hWeWN2WXRhMytUbWIrMmxnVitFZU8zUXJNZDVRc2hHaC9ldgpRQ0x1VFBaaExRK2diKzJML3o3d1JEZGZJSVUzNlpUWHN2UW11NytGOHhvazBwdVgvMHFXYzVkMWZwZmN0QmI5Cm9RcWorYlFia2VaUnNWemJyM0R3THd5cGRUSElMMXB4b1JpZ0FmOU9kZjIxWld3NU14OUh1aE1EZGozQnJaSW8KNGo2TDRST3hBb0dCQVBnNThHL3E4Vy9ocElhV2FVWncwOWZ0bktKdGN5YmdSTHNrakUrYTJEQlJuc0IwZFpUMApSUnQwN21SM3lrRlJqTDRpRFZKV0o3RlNiRi9mTThkaDE2R1dhd0gySFF1dzA3T2YyaDArY3BEbnpBVW15eHdhCnI2aFcwa2swa1BsRU1Md3pkSStPZ2pUR2pwSFpkS0JXNDJYVEhrRThSbUhHa3Z2ckNnbWNHUzVUQW9HQkFNR0MKeC9rakZhdG1YNGI2VmljU3dINEZJRkp2K0IyUkZHcEhlWVBzNkNGZnl6ZDZhT1FpdERZRmw5SFZ1dkNLOCtpNApldk8wOFRINTZiTkt1b0l4STRrb1dXKy9USDJZdnZiWmh4ZnNCOXZvZHd3a050VERkM0xkZ3NxVjRkOE5uVzFECmMxTTZwTmN2QWYzbnA2ZTRja1pPeEZlbm9nL3prQ1NjYXV1ZHNzcVhBb0dCQUlWZk5EWWVLWDJ3NHFMQi9HeXgKd05GN1JUYnMwZnRvY3VXT3ZLa2xETlhoZFFmNVN4Z3VxMWNXVytTMWpJSE9IRzRvZzduaXVzOEc5MTRINHAxVgpqL090aTI5eFdxZm1hY2xRc1hRWEVwMlRQa2JpVWkyV2xSMUJFUm51V1pybExiNWhjMWFTL3BqMnVySFpBbFlJCmVZUkRqRFVHK3RZY1NPdDhuVmNPZldVckFvR0JBSnlCTlZLZHVZemlReWhibHdvSzlIdjNrV2lhL2tTTGJjTjMKQXlSYm1uSkRrNHZQQ0w1YUF2NGVEK2U1d0hPRExqbHQycUw0R0lxZTRSQU14RU0va3NuL1JEWkl0WWh1M1NPdwpVcjVhT3hEME5qUXZwZGoyTGNsTXRkYzYyVzhaQjlXeTA3YmdteWtkNzdKMmhQWDFDeGNwNVVvdlhiY2crZGRpCldtY3RKOGtKQW9HQVJrUTFEK2pyYjM2eGRyR25UeERMcEQ5YUU1SklvRXpkMzlPc2c4UzZLMDh0VEdpWnRPKzMKWEJ4bS96dVpJS3NnQ2ZlSjZLT3hMMEUzMmtLY3hLSG9RTC85dWkxUnViYlRPMzE4ZC9YbFV6M21SZnlGYWNYMwppQlZuaDlQY3owWHQ3YkwvTnpwd3cxZ2owVlBpMDQvWkcyTTB1WWFOcFVjZFdobzhXa1FPdUxzPQotLS0tLUVORCBSU0EgUFJJVkFURSBLRVktLS0tLQo=
+    client-certificate-data: LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUM4akNDQWRvQ0NRQ0lGRGVzZGhSQzNEQU5CZ2txaGtpRzl3MEJBUXNGQURBbU1TUXdJZ1lEVlFRRERCdHYKY0dWdWMyaHBablF0YzJsbmJtVnlRREUxTkRFMU5UQTNORFl3SGhjTk1UZ3hNVEEzTURBek16QTJXaGNOTWpBeApNVEEyTURBek16QTJXakJRTVJjd0ZRWURWUVFLREE1emVYTjBaVzA2YldGemRHVnljekVlTUJ3R0ExVUVDZ3dWCmMzbHpkR1Z0T21Oc2RYTjBaWEl0WVdSdGFXNXpNUlV3RXdZRFZRUUREQXh6ZVhOMFpXMDZZV1J0YVc0d2dnRWkKTUEwR0NTcUdTSWIzRFFFQkFRVUFBNElCRHdBd2dnRUtBb0lCQVFEQ0pFVGpZVGMyVjRQNEQ1aWhFR1ZPVTRDMAp3OVpuT2RhUXJ5eGtFdEZxVFNWSlR6NEdnUmhRLy9OYVA5cUhnRUY5ZU4wSkliRWgwbmx4S0Y2WHVtZXVDamZMClR0a0d1NXFSWTRxb2Y0TG5yRkltNFByNTNkMldrZzV3UVFIYmtvN1VlZEZ1V284dWF6NWNYWVcyM0dnakJkQkQKV1E2Mkl5cCtvaG9KT2ViaHZVVkZZeHBQWTNxM0Fua1BYT2kwQitLMVU0OWNQQWdvU1REQW1wUVF0U3A2Nk1JZQpxeHdkMzErOHJVZXJHZzR1S2ZmRWZ3YVlEbmxtSTZad0hTUVNrVStzZWpNcFZPRStjYzdvWUZsR1RlUVpWeS9OCkJlWlN4Z0xtbXI1ZHhDTHFHZnBGQ1owNXdtVUw3YnA3N0ZCTjZMdEtBemlsbTA0dm0xUmpiSjdqWVBpYkFnTUIKQUFFd0RRWUpLb1pJaHZjTkFRRUxCUUFEZ2dFQkFJanZINDNsTlNiVUp4aUFUdm5DQ0hBZG05OW5HUDVseGU4OQpTM2lxVTMxalROdklxTlUwOUhIV0QzSUdjT2lQOGREMC94dWxZUGI5UDJZcVYwenhHb2ZLSExRL1ZEaldENmdqCjZKc0dNWmdZSTdhdGZBaXNIeER5K090bmxDWDNVdU0xRmNqRm12THhObkoraE05U0RLcjRkT3JYV3diSUFwT0wKd3VaZ3l0a0ZkV1R2d243bDMrTWJnUmNKSjNhSW5iald2eGhwcUZGdC9ybHlNN0Q5RlhNUVRyQ0JTRVVPU0pzSgpZT3QvU29tdlFrVnNWOWd6NWF0UmRFa2hTdFdzaXFFY2RRZkVyWHJiSktBNlc4dkZqdlpDZlFRbWhKei9TMXgrCmx3TTVXR3FRdlZwVGZtMGQ3TkUrWS8rdTM5Y3V2ZnVyeitLbWpOdmxJWHB0R0ZIVGd6az0KLS0tLS1FTkQgQ0VSVElGSUNBVEUtLS0tLQo=
+    client-key-data: LS0tLS1CRUdJTiBSU0EgUFJJVkFURSBLRVktLS0tLQpNSUlFcFFJQkFBS0NBUUVBd2lSRTQyRTNObGVEK0ErWW9SQmxUbE9BdE1QV1p6bldrSzhzWkJMUmFrMGxTVTgrCkJvRVlVUC96V2ovYWg0QkJmWGpkQ1NHeElkSjVjU2hlbDdwbnJnbzN5MDdaQnJ1YWtXT0txSCtDNTZ4U0p1RDYKK2QzZGxwSU9jRUVCMjVLTzFIblJibHFQTG1zK1hGMkZ0dHhvSXdYUVExa090aU1xZnFJYUNUbm00YjFGUldNYQpUMk42dHdKNUQxem90QWZpdFZPUFhEd0lLRWt3d0pxVUVMVXFldWpDSHFzY0hkOWZ2SzFIcXhvT0xpbjN4SDhHCm1BNTVaaU9tY0Iwa0VwRlBySG96S1ZUaFBuSE82R0JaUmsza0dWY3Z6UVhtVXNZQzVwcStYY1FpNmhuNlJRbWQKT2NKbEMrMjZlK3hRVGVpN1NnTTRwWnRPTDV0VVkyeWU0MkQ0bXdJREFRQUJBb0lCQUVBOEo5N2o1NVMvV0JiWApjalM0YkF5Q2JHOWk3S215NVY5eDFrbzJQN3ZnVXBueThoN016L0xjL1BPN2RONm9mUUs2M3NlS3NrVHlrdVB4CjYzcGdYcldqQTlyczk3bldrTW5TR2FxdXBlMXBnclVDYzRnQTlMUkJGeTNsRkd4R0RhUVh1a2U5QnVyZzZQMCsKcEdOZ2djUTB4RGZFVzYyLzdPdUJUd1FHZUhIdnpoclVTdHR6RzRSVlRPOCtGTjVkVHRVTkpleEd6NWZDbVZULwo5QjltbWNOK1AzdVFZRGZuU3M3MXhiVWdiMFNNOU40aGdCSEdyMm9DOTNydmlac3Y2a0w5cTV1WWlLVUgrRHVSCi9peDdEVEE3eXgwaCtNOEt2cHZIY1d5bnFLbnlNeCtYbTFxQlUvMk4vZjhUZnhiRjA0UktSWFM5QXEzK0NKT1YKcXY4V1NlRUNnWUVBN0JrNVB3SGFwSXZyNDROZFdsQWQ2c3pNRmV6d2tDY25uTGRQcFJ1ZTYvWXdxT0RWTmQwdAo4d0JyekcweVYyMHdPWEJZaVJRWC94TWVSUzRRclB0aVVOWkYzakt4Z1hqSkU1elpoVFNFWGZPWStuNmg0bURoCkU3K3o3aHVHTkxkZHNpWlJTMWtDVmI1SllFQnl4NllHVm8yTkZTZndNQ2trMCtFWkVHb05iODhDZ1lFQTBvR28KVFUzWGhnbEVML1pQZ1RXbis1QmMxK21rbE9BQVhmaEc1L25UQi9ORkV0aE9YcXVFM0ZpREtXcUJuSWh5TWMwMApRTytadDJCYmM0TFo3ZXVzS3lWQi93QlJQOGg4TzE4TkVLRGNXTG9EbG04KzMrYTFIODlURlZpUTR4VG5TamxTCmJHU0xMOXFJeDNMQmtsdHJGRW9FWnppeFQzcjBtemY0UGhjVjhYVUNnWUVBeXlraFU1QjYzaUR4Q01PZmJ2bUYKNU9mNXJkeEhYU1AwT0N5ZFFrdzBRb09FZDYyUWlsc2JFUUdKd1UwdGhvR1VmN2ltMXI0TnNCWjVjN29oNjFnZAp6OFd5TlhPVnZ2MTZ1WCs4eFp6ZFVpTXR2azk5YWRsSGxNRGZTY014aHVkS251UE9ESEk1U2lNRjdPZElYWGdGCitEQ3grRVIzc0ZtWE1TTi9kOXBxV1kwQ2dZRUFyRjBhRCtJekhUOHRKQkNNM1FHd1FGM2VoN1I1M0tUbDZBdkoKRTEwT0hLNW0ycDFlM2ttWENrMDNhSXdiQUs4QXdpS0NMTUdEbmhFNkVJVU9LTEFXVTJXNGVqSXFWUTIyRzI4aQo1WUN4cTd6ei94T3I2ZTJnVlNHUHFDVFN6ZjFwdk83ZXZINC9NYm5BMFg3M1hQMloyZzBaT2VQRDI0NlVCeHgzCkdYQVJ1WmtDZ1lFQTR3UTMxU04xN3dTeXZ0S2ZERStDWFRWcTloeXdJT1Q1Z1llSUxIRkxLdG55Y2U4ajhlc0gKR2ZoSVIzeTdRY0pDUWEvTjlacmJJM2VlQlNqalE0ckxvZWIrWW13ZW9vL29Ma1JNMk5mbjQrbzR5ZnZFZjhBTAo2M2o4NC80TkRacE1tVE9yZ3pXM3o4YjhocWg5TWIzZDVVa2Foc1pJTFdVeHllRGxuclF2WENvPQotLS0tLUVORCBSU0EgUFJJVkFURSBLRVktLS0tLQo=
diff --git a/bots/images/openshift b/bots/images/openshift
index 523ab6780..9a5e0b416 120000
--- a/bots/images/openshift
+++ b/bots/images/openshift
@@ -1 +1 @@
-openshift-3b0708e66c1056f2fea3649d5bc266e6e324c88c2890558855958c0ecd2cdde9.qcow2
\ No newline at end of file
+openshift-802170e07fb2ad082ad0ac496b59b2c7537295ca71c79dab2662844a2ba32333.qcow2
\ No newline at end of file
diff --git a/bots/naughty/rhel-x/9722-firewalld-bad-rule-3 b/bots/naughty/rhel-x/9722-firewalld-bad-rule-3
new file mode 100644
index 000000000..f7649f54d
--- /dev/null
+++ b/bots/naughty/rhel-x/9722-firewalld-bad-rule-3
@@ -0,0 +1,5 @@
+Traceback (most recent call last):
+  File "test/verify/check-networking-firewall", line *, in testNetworkingPage
+    m.execute("systemctl stop firewalld")
+*
+RuntimeError: Timed out on 'systemctl stop firewalld'
diff --git a/pkg/networkmanager/firewall-client.es6 b/pkg/networkmanager/firewall-client.es6
index f00910ec1..012f77818 100644
--- a/pkg/networkmanager/firewall-client.es6
+++ b/pkg/networkmanager/firewall-client.es6
@@ -32,11 +32,66 @@ var firewall = {
 cockpit.event_target(firewall);
 
 const firewalld_service = service.proxy('firewalld');
-const firewalld_dbus = cockpit.dbus('org.fedoraproject.FirewallD1');
+var firewalld_dbus = null;
+
+function initFirewalldDbus() {
+    firewalld_dbus = cockpit.dbus('org.fedoraproject.FirewallD1');
+
+    firewalld_dbus.addEventListener('owner', (event, owner) => {
+        firewall.enabled = !!owner;
+
+        firewall.services = {};
+        firewall.enabledServices = new Set();
+
+        if (!firewall.enabled) {
+            firewall.dispatchEvent('changed');
+            return;
+        }
+
+        firewalld_dbus.call('/org/fedoraproject/FirewallD1',
+                            'org.fedoraproject.FirewallD1.zone',
+                            'getServices', [''])
+                .then(reply => fetchServiceInfos(reply[0]))
+                .then(services => services.map(s => firewall.enabledServices.add(s.id)))
+                .then(() => firewall.dispatchEvent('changed'))
+                .catch(error => console.warn(error));
+    });
+
+    firewalld_dbus.subscribe({
+        interface: 'org.fedoraproject.FirewallD1.zone',
+        path: '/org/fedoraproject/FirewallD1',
+        member: 'ServiceAdded'
+    }, (path, iface, signal, args) => {
+        const service = args[1];
+
+        fetchServiceInfos([service])
+                .then(info => {
+                    firewall.enabledServices.add(info[0].id);
+                    firewall.dispatchEvent('changed');
+                })
+                .catch(error => console.warn(error));
+    });
+
+    firewalld_dbus.subscribe({
+        interface: 'org.fedoraproject.FirewallD1.zone',
+        path: '/org/fedoraproject/FirewallD1',
+        member: 'ServiceRemoved'
+    }, (path, iface, signal, args) => {
+        const service = args[1];
+
+        firewall.enabledServices.delete(service);
+        firewall.dispatchEvent('changed');
+    });
+}
 
 firewalld_service.addEventListener('changed', () => {
     let installed = !!firewalld_service.exists;
 
+    /* HACK: cockpit.dbus() remains dead for non-activatable names, so reinitialize it if the service gets enabled and started
+     * See https://github.com/cockpit-project/cockpit/pull/9125 */
+    if (!firewall.enabled && firewalld_service.state == 'running')
+        initFirewalldDbus();
+
     if (firewall.installed == installed)
         return;
 
@@ -76,51 +131,7 @@ function fetchServiceInfos(services) {
     });
 }
 
-firewalld_dbus.addEventListener('owner', (event, owner) => {
-    firewall.enabled = !!owner;
-
-    firewall.services = {};
-    firewall.enabledServices = new Set();
-
-    if (!firewall.enabled) {
-        firewall.dispatchEvent('changed');
-        return;
-    }
-
-    firewalld_dbus.call('/org/fedoraproject/FirewallD1',
-                        'org.fedoraproject.FirewallD1.zone',
-                        'getServices', [''])
-            .then(reply => fetchServiceInfos(reply[0]))
-            .then(services => services.map(s => firewall.enabledServices.add(s.id)))
-            .then(() => firewall.dispatchEvent('changed'))
-            .catch(error => console.warn(error));
-});
-
-firewalld_dbus.subscribe({
-    interface: 'org.fedoraproject.FirewallD1.zone',
-    path: '/org/fedoraproject/FirewallD1',
-    member: 'ServiceAdded'
-}, (path, iface, signal, args) => {
-    const service = args[1];
-
-    fetchServiceInfos([service])
-            .then(info => {
-                firewall.enabledServices.add(info[0].id);
-                firewall.dispatchEvent('changed');
-            })
-            .catch(error => console.warn(error));
-});
-
-firewalld_dbus.subscribe({
-    interface: 'org.fedoraproject.FirewallD1.zone',
-    path: '/org/fedoraproject/FirewallD1',
-    member: 'ServiceRemoved'
-}, (path, iface, signal, args) => {
-    const service = args[1];
-
-    firewall.enabledServices.delete(service);
-    firewall.dispatchEvent('changed');
-});
+initFirewalldDbus();
 
 cockpit.spawn(['sh', '-c', 'pkcheck --action-id org.fedoraproject.FirewallD1.all --process $$ --allow-user-interaction 2>&1'])
         .done(() => {
diff --git a/src/ssh/cockpitsshknownhosts.c b/src/ssh/cockpitsshknownhosts.c
index 7c33e94e1..334effd4d 100644
--- a/src/ssh/cockpitsshknownhosts.c
+++ b/src/ssh/cockpitsshknownhosts.c
@@ -456,3 +456,23 @@ cockpit_is_host_known (const gchar *known_hosts_file,
   /* we did not find anything, end of file*/
   return ret;
 }
+
+enum ssh_known_hosts_e
+ssh_session_is_known_server (ssh_session session)
+{
+    enum ssh_server_known_e result;
+
+    result = ssh_is_server_known (session);
+
+    switch(result)
+      {
+          case SSH_SERVER_ERROR: return SSH_KNOWN_HOSTS_ERROR;
+          case SSH_SERVER_FILE_NOT_FOUND: return SSH_KNOWN_HOSTS_NOT_FOUND;
+          case SSH_SERVER_NOT_KNOWN: return SSH_KNOWN_HOSTS_UNKNOWN;
+          case SSH_SERVER_KNOWN_OK: return SSH_KNOWN_HOSTS_OK;
+          case SSH_SERVER_KNOWN_CHANGED: return SSH_KNOWN_HOSTS_CHANGED;
+          case SSH_SERVER_FOUND_OTHER: return SSH_KNOWN_HOSTS_OTHER;
+      }
+
+    g_assert_not_reached ();
+}
diff --git a/src/ssh/cockpitsshknownhosts.h b/src/ssh/cockpitsshknownhosts.h
index 755610321..cb01a6b0a 100644
--- a/src/ssh/cockpitsshknownhosts.h
+++ b/src/ssh/cockpitsshknownhosts.h
@@ -26,7 +26,15 @@
 
 G_BEGIN_DECLS
 
-#define SSH_KNOWN_HOSTS_OK TRUE
+/* translate to old ssh_server_known_e enum for deprecated ssh_is_server_known() API */
+enum ssh_known_hosts_e {
+    SSH_KNOWN_HOSTS_ERROR = -2,
+    SSH_KNOWN_HOSTS_NOT_FOUND = -1,
+    SSH_KNOWN_HOSTS_UNKNOWN = 0,
+    SSH_KNOWN_HOSTS_OK,
+    SSH_KNOWN_HOSTS_CHANGED,
+    SSH_KNOWN_HOSTS_OTHER,
+};
 
 void            shim_set_knownhosts_file                   (const gchar *file);
 
@@ -39,6 +47,7 @@ gboolean        cockpit_is_host_known                      (const gchar *known_h
                                                             const gchar *host,
                                                             guint port);
 
+enum            ssh_known_hosts_e ssh_session_is_known_server (ssh_session session);
 
 G_END_DECLS
 
diff --git a/src/ssh/cockpitsshrelay.c b/src/ssh/cockpitsshrelay.c
index d47a0b65d..196ee9391 100644
--- a/src/ssh/cockpitsshrelay.c
+++ b/src/ssh/cockpitsshrelay.c
@@ -684,7 +684,7 @@ verify_knownhost (CockpitSshData *data,
   const gchar *ret = "invalid-hostkey";
   ssh_key key = NULL;
   unsigned char *hash = NULL;
-  int state;
+  enum ssh_known_hosts_e state;
   gsize len;
 
   g_warn_if_fail (ssh_session_export_known_hosts_entry(data->session, &data->host_key) == SSH_OK);
@@ -742,14 +742,14 @@ verify_knownhost (CockpitSshData *data,
     }
 #endif
 
-  state = ssh_is_server_known (data->session);
-  if (state == SSH_SERVER_KNOWN_OK)
+  state = ssh_session_is_known_server (data->session);
+  if (state == SSH_KNOWN_HOSTS_OK)
     {
       g_debug ("%s: verified host key", data->logname);
       ret = NULL; /* success */
       goto done;
     }
-  else if (state == SSH_SERVER_ERROR)
+  else if (state == SSH_KNOWN_HOSTS_ERROR)
     {
       g_warning ("%s: couldn't check host key: %s", data->logname,
                  ssh_get_error (data->session));
@@ -759,22 +759,22 @@ verify_knownhost (CockpitSshData *data,
 
   switch (state)
     {
-    case SSH_SERVER_KNOWN_OK:
-    case SSH_SERVER_ERROR:
+    case SSH_KNOWN_HOSTS_OK:
+    case SSH_KNOWN_HOSTS_ERROR:
       g_assert_not_reached ();
       break;
-    case SSH_SERVER_KNOWN_CHANGED:
+    case SSH_KNOWN_HOSTS_CHANGED:
       g_message ("%s: %s host key for server has changed to: %s",
                  data->logname, data->host_key_type, data->host_fingerprint);
       break;
-    case SSH_SERVER_FOUND_OTHER:
+    case SSH_KNOWN_HOSTS_OTHER:
       g_message ("%s: host key for this server changed key type: %s",
                  data->logname, data->host_key_type);
       break;
-    case SSH_SERVER_FILE_NOT_FOUND:
+    case SSH_KNOWN_HOSTS_NOT_FOUND:
       g_debug ("%s: Couldn't find the known hosts file", data->logname);
       /* fall through */
-    case SSH_SERVER_NOT_KNOWN:
+    case SSH_KNOWN_HOSTS_UNKNOWN:
       ret = prompt_for_host_key (data);
       if (ret)
         {
diff --git a/test/verify/check-networking-firewall b/test/verify/check-networking-firewall
index 46ffffadd..f2a0d0919 100755
--- a/test/verify/check-networking-firewall
+++ b/test/verify/check-networking-firewall
@@ -35,34 +35,40 @@ def wait_unit_state(machine, unit, state):
 
 
 @skipImage("firewalld not installed", "fedora-atomic", "rhel-atomic", "continuous-atomic")
+@skipImage("enabling firewall got fixed in PR #10511", "rhel-7-6-distropkg")
 class TestFirewall(MachineCase):
 
     def testNetworkingPage(self):
         b = self.browser
         m = self.machine
 
+        m.execute("systemctl stop firewalld")
+
         self.login_and_go("/network")
         b.wait_visible("#networking-firewall")
-
-        m.execute("systemctl stop firewalld")
         b.wait_in_text("#networking-firewall-switch label.active", "Off")
-        m.execute("systemctl start firewalld")
-        b.wait_in_text("#networking-firewall-switch label.active", "On")
 
         # to toggle the switch, click on the non-active label, where the slider is
+        b.click("#networking-firewall-switch label:not(.active)")
+        b.wait_present("#networking-firewall-switch label.active:not(.disabled)")
+        b.wait_in_text("#networking-firewall-switch label.active", "On")
+        wait_unit_state(m, "firewalld", "active")
+        active_rules = m.execute("firewall-cmd --list-services").split()
+        b.wait_in_text("#networking-firewall-summary", "{} Active Rule".format(len(active_rules)))
+
         b.click("#networking-firewall-switch label:not(.active)")
         b.wait_present("#networking-firewall-switch label.active:not(.disabled)")
         b.wait_in_text("#networking-firewall-switch label.active", "Off")
         wait_unit_state(m, "firewalld", "inactive")
         b.wait_in_text("#networking-firewall-summary", "0 Active Rules")
 
-        b.click("#networking-firewall-switch label:not(.active)")
-        b.wait_present("#networking-firewall-switch label.active:not(.disabled)")
+        # toggle the service from CLI, page should react
+        m.execute("systemctl start firewalld")
         b.wait_in_text("#networking-firewall-switch label.active", "On")
-        wait_unit_state(m, "firewalld", "active")
-
-        active_rules = m.execute("firewall-cmd --list-services").split()
         b.wait_in_text("#networking-firewall-summary", "{} Active Rule".format(len(active_rules)))
+        m.execute("systemctl stop firewalld")
+        b.wait_in_text("#networking-firewall-switch label.active", "Off")
+        b.wait_in_text("#networking-firewall-summary", "0 Active Rules")
 
         b.click("#networking-firewall-link")
         b.enter_page("/network/firewall")
@@ -70,13 +76,20 @@ class TestFirewall(MachineCase):
         b.click(".breadcrumb li:first a")
         b.enter_page("/network")
 
-
     def testFirewallPage(self):
         b = self.browser
         m = self.machine
 
+        m.execute("systemctl stop firewalld")
         self.login_and_go("/network/firewall")
-        m.execute("systemctl start firewalld")
+
+        # to toggle the switch, click on the non-active label, where the slider is
+        b.wait_present(".btn-onoff-ct label.active")
+        b.wait_in_text(".btn-onoff-ct label.active", "Off")
+        b.click(".btn-onoff-ct label.active span")
+        b.wait_present(".btn-onoff-ct label.active:not(.disabled)")
+        b.wait_in_text(".btn-onoff-ct label.active", "On")
+        wait_unit_state(m, "firewalld", "active")
 
         # ensure that pop3 is not enabled (shouldn't be on any of our images),
         # so that we can use it for testing
@@ -97,6 +110,12 @@ class TestFirewall(MachineCase):
         b.wait_not_present("tr[data-row-id='pop3']")
         self.assertNotIn('pop3', m.execute("firewall-cmd --list-services").split())
 
+        # switch service off again
+        b.click(".btn-onoff-ct label.active span")
+        b.wait_present(".btn-onoff-ct label.active:not(.disabled)")
+        b.wait_in_text(".btn-onoff-ct label.active", "Off")
+        wait_unit_state(m, "firewalld", "inactive")
+
 
     def testAddServices(self):
         b = self.browser
diff --git a/test/verify/machineslib.py b/test/verify/machineslib.py
index 5f683f143..a40e88cf6 100755
--- a/test/verify/machineslib.py
+++ b/test/verify/machineslib.py
@@ -151,7 +151,7 @@ class TestMachines(MachineCase):
         m.execute("sed -i 's/\"priority\".*$/\"priority\": 100,/' {0}".format("/usr/share/cockpit/machines/manifest.json"))
         m.execute("[ ! -e {0} ] || sed -i 's/\"priority\".*$/\"priority\": 0,/' {0}".format("/usr/share/cockpit/ovirt/manifest.json"))
         # we don't have configuration to open the firewall for local libvirt machines, so just stop firewalld
-        m.execute("systemctl stop firewalld")
+        m.execute("systemctl stop firewalld; systemctl try-restart libvirtd")
 
     def startLibvirt(self):
         m = self.machine
```
</details>

martinpitt commented:
Rebased to fix (simple) conflict, and fixed testNetworkSettings.

martinpitt commented:
This [debian-testing failure](https://fedorapeople.org/groups/cockpit/logs/pull-10512-20181108-034545-51ace850-verify-debian-testing/log.html#249) looks exactly like on rhel-x; similar firewalld/libvirt combination presumably.

martinpitt commented:
The last failure is [check-machines-dbus](https://fedorapeople.org/groups/cockpit/logs/pull-10512-20181108-033030-51ace850-verify-ubuntu-stable/log.html#228) on ubuntu-stable. This is persistent and also reproducible locally.

martinpitt commented:
Fixed. I really hope I've got everything now..

martinpitt commented:
Hooray, all green at last!

martinpitt commented on bots/images/scripts/debian.setup:
```diff
@@ -75,8 +77,9 @@ case "$RELEASE" in
-    testing)
-        COCKPIT_DEPS="${COCKPIT_DEPS} libvirt-dbus"
+    bionic)
+        # thesse packages are not in Ubuntu 18.04
```
Thanks, fixed.
